loaders/src/glTF/2.0/Extensions/MSFT_audio_emitter.ts

```diff
@@ -0,0 +1,459 @@
+                    }
+
+                    return Promise.all(promises).then(() => {});
+                });
+            });
+        }
+
+        protected _loadNodeAsync(context: string, node: _ILoaderNode): Nullable<Promise<void>> { 
+            return this._loadExtensionAsync<_IEmittersReference>(context, node, (extensionContext, extension) => {
+                return this._loader._loadNodeAsync(extensionContext, node).then(() => {
+
+                    const promises = new Array<Promise<void>>();
+                    _ArrayItem.Assign(this._emitters);
+                    for (const emitterIndex of extension.emitters) {
+                        const emitter = GLTFLoader._GetProperty(extensionContext, this._emitters, emitterIndex);
+                        promises.push(this._loadEmitterAsync(`#/extensions/${this.name}/emitter/${emitter._index}`, emitter).then(() => {
+                            if (node._babylonMesh) {
+                                for (const sound of emitter._babylonSounds) {
+                                    sound.attachToMesh(node._babylonMesh);
+                                    if (emitter.innerAngle != undefined || emitter.outerAngle != undefined) {
+                                        sound.setLocalDirectionToMesh(new Vector3(0, 0, 1));
+                                        sound.setDirectionalCone(2 * Tools.ToDegrees(emitter.innerAngle == undefined ? Math.PI : emitter.innerAngle),
+                                                                 2 * Tools.ToDegrees(emitter.outerAngle == undefined ? Math.PI : emitter.outerAngle), 0);
+                                    }
+                                }
+                            }
+                        }));
+                    }
+
+                    return Promise.all(promises).then(() => {});
+                });
+            });
+        }
+
+        protected _loadAnimationAsync(context: string, animation: _ILoaderAnimation): Nullable<Promise<void>> { 
+            return this._loadExtensionAsync<_ILoaderAnimationEvents>(context, animation, (extensionContext, extension) => {
+                return this._loader._loadAnimationAsync(extensionContext, animation).then(() => {
+                    const promises = new Array<Promise<void>>();
+                    let babylonAnimationGroup = animation._babylonAnimationGroup;
+
+                    _ArrayItem.Assign(extension.events);
+                    for (const event of extension.events) {
+                        promises.push(this._loadAnimationEventAsync(`${context}/extension/${NAME}/events/${event._index}`, context, animation, event, babylonAnimationGroup!));
+                    }
+
+                    return Promise.all(promises).then(() => {
+                        // Make sure all audio stops when the animation is terminated.
+                        const getCallback = (pause: Boolean) => {
```

Comment:
this is inefficient and IMO harder to read than if you just repeat the for loop

Reply:
why is it inefficient? I need to capture the current value of the emitter list, if I don't later when these care called it will be empty.

Reply:
oh you mean testing the conditional each loop iteration

Reply:
might as well not bother with this function and just inline it?